### DISCUSSION TITLE
Issue interpreting the time value for monthly climatologies

### DISCUSSION THREAD

**Discussion by arnemelsom** [cf-convention/discuss#456] [Q&A about using CF] [answered]
With reference to Example 7.10 on [https://cfconventions.org/cf-conventions/cf-conventions.html](https://cfconventions.org/cf-conventions/cf-conventions.html) I don't understand how e.g. a time value of '1965-1-15' emerges for the January climatology with bounds "1961-1-1", "1970-2-1". There are 10 January months in this period, and the central time is sometime in July 1965. How is the time value properly defined in this example?

**Comment by JonathanGregory** [comment]
Dear @arnemelsom 

Thanks for the question. For a climatological mean, the text of the convention (a bit earlier, in the section containing that example) says, "The time coordinates should be values that are representative of the climatological time intervals, such that an application which does not recognise climatological time will nonetheless be able to make a reasonable interpretation." An application which is unaware of climatological bounds would presumably examine the time coordinates to see what the data applies to. For that reason, a date within January is chosen in order to indicate it's a climatological mean for January. Choosing the middle of the month, and the first year, is arbitrary. The full information can't be indicated by a simple coordinate value, but "January" would be a "reasonable interpretation". Does that make sense?

Best wishes

Jonathan

**Comment by larsbarring** [comment]
For this particular example it becomes more clear if you consider that the **input data** for the first climatological average is data for 10 January months, like this:
``` txt
time = "1961-01-15", 1962-01-15" .... "1969-01-15", "1970-01-15";
time_bounds = "1961-01-01 00", "1961-02-01 00",
              "1962-01-01 00", "1962-02-01 00"
              ....
              "1969-01-01 00", "1969-02-01 00",
              "1970-01-01 00", "1970-02-01 00"
```
I explicitly included the hour in the bounds to show why 1 February is at the end of the bounds even though it is implied when the hour is 0. As Jonathan explains the time [points] can be anything within the bounds depending on what best represent the data. E.g. for daily observed precipitation it is common to have the time [point] at the end of the corresponding time bounds, because that is how precipitation is observed.

From this it is more clear why the climatological average of the input data is "1965-01-15" in the example.  It really does not make sense to take the midpoint between the start and end of a climatological bounds. Instead one have to consider the `cell_method`, which in this example is:
> precipitation: cell_methods="time: sum within years time: mean over years" 

where the `mean over years` relates to the *year* part of the bounds: `1961` ... `1970`, which gives the average 1965
and the `sum within years` relates to the *month and day* part of the bounds `01-01 00`, `02-01 00`, which gives the average 01-15 (as a reasonable approximation) which is consistent with the month and day part of the time [point]. As this is the average total precipitation for the whole month, it might have been relevant have the time [points] at the end of the bounds (or the last day of the month), i.e. `1965-01-31`, but this is a matter of context and perhaps taste.

**Comment by arnemelsom** [comment, marked as answer]
Dear Jonathan, Lars,
Thanks for taking the time to answer my inquiry, I really appreciate that!

From the beginning, my view was that the essential information in Example 7.10 is the values of 'climatology_bounds', and that the value of 'time' basically is superfluous. So my inclination was to scrap 'time' all together. But I guess considerations of how other users (and/or software) would interpret the file contents suggest that a 'time' variable/value should be included in the climatology. Which is why I wanted to understand its definition.

Your explanation makes good sense (given that years are integers, so integer division rules apply to averaging). I've received a data set with daily averages, which lacked time bounds values. So I added the appropriate time bounds attribute, variable and values. I found out that nco's ncra can perform the averaging, leading to a result for time specification according to your explanations (using the --clm_bnd run time command line argument). Actually, I was a bit surprised to see that the result was given a time of 01-15, as 01-16 would be the average time for my daily input. So I guess that mm-15 is applied 
for all months by definition.

Arne

PS. I'm well aware that time given as yyyy-mm-dd has a complete specification of yyyy-mm-ddT00:00:00.0. But good that you explicitly stated this, should someone without this understanding stumble upon this thread.
